pkg/ebpf/uprobes/attacher.go

```diff
@@ -319,7 +335,7 @@ type UprobeAttacher struct {
 // that depend on the configuration, so any changes to the configuration after the
 // attacher would make those caches incoherent. This way we ensure that the attacher
 // is always consistent with the configuration it was created with.
-func NewUprobeAttacher(name string, config AttacherConfig, mgr ProbeManager, onAttachCallback AttachCallback, inspector BinaryInspector) (*UprobeAttacher, error) {
+func NewUprobeAttacher(name string, config AttacherConfig, mgr ProbeManager, onAttachCallback AttachCallback, inspector BinaryInspector, processMonitor ProcessMonitor) (*UprobeAttacher, error) {
```

Comment:
I think this ctor is a bit overloaded with the params, perhaps we can introduce some struct to contain a few of those params together (can be done in a separate PR as well)

Reply:
Felt the same way, maybe a "dependencies" struct to hold the manager, inspector and monitor